src/ec2/instanceStateManager.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Ec2Client } from '../shared/clients/ec2Client'
+import { ToolkitError, isAwsError } from '../shared/errors'
+import { showMessageWithCancel } from '../shared/utilities/messages'
+import { Timeout } from '../shared/utilities/timeoutUtils'
+import { Ec2Selection } from './utils'
+
+export class InstanceStateManager {
+    private readonly client: Ec2Client
+
+    public constructor(protected readonly instanceId: string, protected readonly regionCode: string) {
+        this.client = this.getEc2Client()
+    }
```

Comment:
This class doesn't really hold any non-trivial state. Can its functions live on `src/shared/clients/ec2Client.ts` ?

The EC2 service is starting to accumulate a lot of classes. As a developer if I want to use the Toolkit's EC2 layer, it's nice to be able to reach for `ec2Client` and just look for functions in a flat list. In contrast, having to hunt around for this or that class is an extra step (or multiple steps if there are many classes).

New classes are useful for non-trivial topics. But when things are trivial, that's a gift, let's keep them as trivial as possible.

Reply:
I was imagining the EC2 client as purely a wrapper of the SDK and thought I should separate the business logic from the SDK logic. I was trying to avoid overloading the EC2 client in general as well. However, I definitely see what you mean about making this harder to find. Going to work on integrating this logic into the EC2Client itself.

Reply:
> I was imagining the EC2 client as purely a wrapper of the SDK

If it's pure then we don't need it :) We need a Toolkit-layer wrapper around the SDK that does useful things, not a passthrough interface.

Reply:
Yeah that makes sense. This seems to be an example of trying to over-modularize, to the point where its difficult to find what you want to use. I just committed the refactor of moving to EC2Client.